src/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,240 @@
+import React from 'react';
+import styled from 'styled-components';
+import { Placement } from '@popperjs/core/lib/enums';
+
+import { theme } from '../../essentials/theme';
+import { get } from '../../utils/themeGet';
+import { Colors, Spaces } from '../../essentials';
+import { ChevronDownIcon, ChevronUpIcon } from '../../icons/index';
+
+import { Text } from '../Text/Text';
+
+import { PopoverContent } from './PopoverContent';
+import { handleKeyDown } from '../../utils/hooks/useHandleKeyDown';
+import { useStateWithTimeout } from '../../utils/hooks/useStateWithTimeout';
+
+interface PopoverProps {
+    /**
+     * Popover Trigger (Only use Text, Link, Button or Icon component from @wave as a trigger)
+     */
+    children: React.ReactNode;
+    /**
+     * Popover content (can be any valid React Element or component containing React Elements)
+     */
+    content: React.ReactNode;
+    /**
+     * Optional: Specify the Popover content placement (placement changes automatically, if Popover content cannot fit inside viewport with selected placement)
+     */
+    placement?: Placement;
+    /**
+     * Optional: Specify the Popover content offset (margin between Popover trigger and content)
+     */
+    offset?: number;
+    /**
+     * Optional: Keep popover content open by default
+     */
+    isOpen?: boolean;
+    /**
+     * Optional: Define a callback for when Popover content is opened
+     */
+    onOpen?: () => void;
+    /**
+     * Optional: Define a callback for when Popover content is closed
+     */
+    onClose?: () => void;
+}
+
+const PopoverTrigger = styled.div.attrs({ theme })`
+    display: inline-flex;
+    border-radius: ${get('radii.2')};
+`;
+
+const DefaultPopoverWrapper = styled.div.attrs({ theme })`
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    border: 1px solid ${get('semanticColors.button.secondary.borderHover')};
+    padding: 0.8125rem ${Spaces[2]};
+    border-radius: ${get('radii.2')};
+
+    &:hover {
+        cursor: pointer;
+        background-color: ${get('semanticColors.background.secondary')} !important;
+    }
+`;
+
+/**
+ * ** Primary UI element for content whose visibility can be toggled with a click on trigger element **
+ *
+ * Renders a UI component that opens a popup dialog on click of a trigger element.
```

Comment:
Why do we need this duplication? It is available in the mdx

Reply:
These were docs format for Storybook, I will remove it